skrifa/src/outline/autohint/latin/hint.rs

```diff
@@ -45,62 +53,64 @@ pub(crate) fn hint_edges(
 fn align_edges_to_blues(
     axis: &mut Axis,
     metrics: &ScaledAxisMetrics,
+    group: ScriptGroup,
     scale: &Scale,
 ) -> Option<usize> {
     let mut anchor_ix = None;
-    // For a vertical axis, begin by aligning stems to blue zones
-    if axis.dim == Axis::VERTICAL {
-        for edge_ix in 0..axis.edges.len() {
-            let edges = axis.edges.as_mut_slice();
-            let edge = &edges[edge_ix];
-            if edge.flags & Edge::DONE != 0 {
-                continue;
-            }
-            let edge2_ix = edge.link_ix.map(|x| x as usize);
-            let edge2 = edge2_ix.map(|ix| &edges[ix]);
-            // If we have two neutral zones, skip one of them
-            if let (true, Some(edge2)) = (edge.blue_edge.is_some(), edge2) {
-                if edge2.blue_edge.is_some() {
-                    let skip_ix = if edge2.flags & Edge::NEUTRAL != 0 {
-                        edge2_ix
-                    } else if edge.flags & Edge::NEUTRAL != 0 {
-                        Some(edge_ix)
-                    } else {
-                        None
-                    };
-                    if let Some(skip_ix) = skip_ix {
-                        let skip_edge = &mut edges[skip_ix];
-                        skip_edge.blue_edge = None;
-                        skip_edge.flags &= !Edge::NEUTRAL;
-                    }
-                }
-            }
-            // Flip edges if the other is aligned to a blue zone
-            let blue = edges[edge_ix].blue_edge;
-            let (blue, edge1_ix, edge2_ix) = if let Some(blue) = blue {
-                (blue, Some(edge_ix), edge2_ix)
-            } else if let Some(edge2_blue) = edge2_ix.and_then(|ix| edges[ix].blue_edge) {
-                (edge2_blue, edge2_ix, Some(edge_ix))
-            } else {
-                (Default::default(), None, None)
-            };
-            let Some(edge1_ix) = edge1_ix else {
-                continue;
-            };
-            let edge1 = &mut edges[edge1_ix];
-            edge1.pos = blue.fitted;
-            edge1.flags |= Edge::DONE;
-            if let Some(edge2_ix) = edge2_ix {
-                let edge2 = &mut edges[edge2_ix];
-                if edge2.blue_edge.is_none() {
-                    edge2.flags |= Edge::DONE;
-                    align_linked_edge(axis, metrics, scale, edge1_ix, edge2_ix);
+    // For default script group, only do vertical blues
+    if group == ScriptGroup::Default && axis.dim != Axis::VERTICAL {
+        return anchor_ix;
+    }
+    for edge_ix in 0..axis.edges.len() {
+        let edges = axis.edges.as_mut_slice();
+        let edge = &edges[edge_ix];
+        if edge.flags & Edge::DONE != 0 {
+            continue;
+        }
+        let edge2_ix = edge.link_ix.map(|x| x as usize);
+        let edge2 = edge2_ix.map(|ix| &edges[ix]);
+        // If we have two neutral zones, skip one of them.
+        if let (true, Some(edge2)) = (edge.blue_edge.is_some(), edge2) {
```

Comment:
can't believe we still don't have if-let chains!? but apparently not (https://github.com/rust-lang/rust/issues/53667)

(if we did, this could be,

```rust
if let Some(edge2) = edge2 && edge.blue_edge.is_some() { .. }
```
)

Reply:
This seems to always be just over the horizon... at least we can avoid the double `if` with the tuple pattern matching so I don't want to complain too much